AutoCollection/ClientRequestParser.ts

```diff
@@ -0,0 +1,119 @@
+///<reference path="..\Declarations\node\node.d.ts" />
+
+import http = require("http");
+import url = require("url");
+
+import ContractsModule = require("../Library/Contracts");
+import Client = require("../Library/Client");
+import Logging = require("../Library/Logging");
+import Util = require("../Library/Util");
+import RequestParser = require("./RequestParser");
+
+/**
+ * Helper class to read data from the requst/response objects and convert them into the telemetry contract
+ */
+class ClientRequestParser extends RequestParser {
+    constructor(requestOptions: string | Object, request: http.ClientRequest) {
+        super();
+        if (request && requestOptions) {
+            // The ClientRequest.method property isn't documented, but is always there.
+            this.method = (<any>request).method;
+
+            this.url = ClientRequestParser._getUrlFromRequestOptions(requestOptions, request);
+            this.startTime = +new Date();
+        }
+    }
+
+    /**
+     * Called when the ClientRequest emits an error event.
+     */
+    public onError(error: Error, properties?: { [key: string]: string }) {
+        this._setStatus(undefined, error, properties);
+    }
+
+    /**
+     * Called when the ClientRequest emits a response event.
+     */
+    public onResponse(response: http.ClientResponse, properties?: { [key: string]: string }) {
+        this._setStatus(response.statusCode, undefined, properties);
+    }
+
+    /**
+     * Gets a dependency data contract object for a completed ClientRequest.
+     */
+    public getDependencyData(): ContractsModule.Contracts.Data<ContractsModule.Contracts.RemoteDependencyData> {
+        let urlObject = url.parse(this.url);
+        urlObject.search = undefined;
+        urlObject.hash = undefined;
+        let dependencyName = this.method.toUpperCase() + " " + url.format(urlObject);
+
+        let remoteDependency = new ContractsModule.Contracts.RemoteDependencyData();
+        remoteDependency.target = urlObject.hostname;
+        remoteDependency.name = dependencyName;
+        remoteDependency.commandName = this.url;
+        remoteDependency.value = this.duration;
+        remoteDependency.success = this._isSuccess();
+        remoteDependency.async = true;
+        remoteDependency.dependencyTypeName =
+            ContractsModule.Contracts.DependencyKind[ContractsModule.Contracts.DependencyKind.Http];
+        remoteDependency.dependencyKind = ContractsModule.Contracts.DependencyKind.Http;
```

Comment:
I believe a new dependency type, something called AI dependency, is being introduced, and that would be used instead of HTTP for communications between AI components.

Check with Nizar on the details.